### ISSUE TITLE
Thorium crashed in this morning. Works well last night.

### ISSUE THREAD

**Issue by ladyrick** [Alex313031/Thorium-MacOS#66]
**System Details**
 - OS: MacOS 15.0.1
 - Thorium Version: M126.0.6478.231

**Problem**
This morning my Thorium just crashed without any error prompts. The browser window suddenly disappeared.
Until last night, it works well for a long time.

**Screenshots**
No screenshots.

**Additional Notes**
I install thorium with brew.
I tried to do a clean uninstall and reinstall. No help.
I tried the previous version of Thorium. No help.
My Mac didn't restart for a long time. No big system update(which requires reboot) installed. I don't know if there are some small updates silently installed.
I tried to reboot. No help.
Safari works well. Google Chrome works well.

**Comment by matgessel** [comment]
Same. Thorium 124 x64 on macOS 10.15.7. Crashes when restoring tabs or after creating 1 - 4 new tabs. Crashes mid-render of page. Just started happening tonight and now happens every time I restore tabs or browse to a website, including the default homepage (DDG. DDG homepage crashes every time; Google homepage hasn't crashed yet for me, but it still crashes when I browse to google.com).

`com.apple.xpc.launchd[1] (org.chromium.Thorium.4768[3765]): Service exited due to SIGTRAP`

Does not crash on chrome://flags, chrome://settings or chrome://extensions. Does not crash when opening a page saved as webpage complete.

Does crash when network is disconnected and I launch Thorium, restore tabs and select a restored tab.

Troubleshooting scenarios tried:

- Disable all extensions
- Deleted entire Thorium directory in Application Support
- reinstall into fresh non-admin account
- backdate system clock several days
- add Thorium to Privacy > Developer Tools (Gatekeeper bypass)

Initial crash occurred 2024.10.26 03:13 UTC.

Most recent software install/update is XProtectPayloads v147 on 2024.10.16.

**Comment by anta40** [comment]
Yeah me too (M2 Sonoma 14.5).
Thorium version: 126.0.6478.231 

I think this issue started to occur last night. Previously no issue.
It'll crash after I open any website (youtube, wikipedia, etc).

Not sure if the cause is browser extensions, because my Chrome also uses the same extensions.

```
 ~ $ thorium
Warning: SetDeviceLostCallback is deprecated. Pass the callback in the device descriptor instead.
[23503:259:1026/232518.840835:ERROR:unexportable_key_mac.mm(348)] Unexportable keys unavailable because keychain-access-group entitlement missing or incorrect. Expected value: .org.chromium.Thorium.devicetrust
Created TensorFlow Lite XNNPACK delegate for CPU.
[23503:34563:1026/232521.295419:ERROR:unexportable_key_mac.mm(348)] Unexportable keys unavailable because keychain-access-group entitlement missing or incorrect. Expected value: .org.chromium.Thorium.unexportable-keys
Attempting to use a delegate that only supports static-sized tensors with a graph that has dynamic-sized tensors (tensor#180 is a dynamic-sized tensor).
[23503:37379:1026/232521.403918:ERROR:registration_request.cc(291)] Registration response error message: DEPRECATED_ENDPOINT
[23503:37379:1026/232545.115794:ERROR:registration_request.cc(291)] Registration response error message: DEPRECATED_ENDPOINT
[1026/232554.518068:WARNING:in_range_cast.h(38)] value -634136515 out of range
[1026/232554.528903:WARNING:crash_report_exception_handler.cc(235)] UniversalExceptionRaise: (os/kern) failure (5)
[23512:14083:1026/232554.530951:ERROR:ssl_client_socket_impl.cc(878)] handshake failed; returned -1, SSL error code 1, net_error -3
[1]    23503 trace trap  thorium
```

**Comment by arkasoft** [comment]
Yeah me too (X86 i9) running latest Sequoia 15.0.1
Everything was working fine and suddenly this morning Thorium crashes without any warning. Its not depent on a site either. I just after selecting a profile it will crash after a little time for me.
Rebooted and ran again, there had not been any updates that I know about either.

**Comment by melMass** [comment]
Happens on silicon too:

```sh
[22968:259:1026/151600.823061:ERROR:unexportable_key_mac.mm(348)] Unexportable keys unavailable because keychain-access-group entitlement missing or incorrect. Expected value: .org.chromium.Thorium.devicetrust
Warning: SetDeviceLostCallback is deprecated. Pass the callback in the device descriptor instead.                            
[22968:41475:1026/151603.253519:ERROR:unexportable_key_mac.mm(348)] Unexportable keys unavailable because keychain-access-group entitlement missing or incorrect. Expected value: .org.chromium.Thorium.unexportable-keys   
Created TensorFlow Lite XNNPACK delegate for CPU.
Attempting to use a delegate that only supports static-sized tensors with a graph that has dynamic-sized tensors (tensor#180 is a dynamic-sized tensor).
[1026/151604.068929:WARNING:crash_report_exception_handler.cc(235)] UniversalExceptionRaise: (os/kern) failure (5)
```

**Comment by Drugoy** [comment]
I get similar error as @melMass on Sonoma 14.6 on M1 CPU.
The browser was working just fine just yesterday.
Today I came to it having crashed.
Opening it - works, but loading any (?) sites results into such a crash.

At first I thought my profile got corrupted, so I tried to create a new one and... it crashes as well even when trying to open google.com

I am deeply worried about the possible reasons of such a crash given it happened to a lot of users AT ONCE.
It's either something like an important certificate getting outdated or... an evil mechanism that allows some 3rd party to control when to break user agents for users.

**Comment by Quotation** [comment]
Same here. Just using Thorium as usual half an hour ago, but now it keeps crashing.
Installing two older versions of Thorium didn’t help; it still crashes.

I checked some websites:
No crashes: apple.com
Crashes: google.com, x.com, github.com, wikipedia.org, amazon.com and more...

**Comment by gothickitty93** [comment]
#67 I think these two are one and the same, to be honest. I'm having the exact same as everyone else. Both my home M2, and my work i7. I also tried setting the time back, but that didn't help. Thorium is working fine on Windows.

**Comment by matgessel** [comment]
A workaround has been posted in #67

**Comment by ladyrick** [comment]
Downgrade to [M119.0.6045.214](https://github.com/Alex313031/Thorium-MacOS/releases/tag/M119.0.6045.214) works.

**Comment by gothickitty93** [comment]
> Downgrade to [M119.0.6045.214](https://github.com/Alex313031/Thorium-MacOS/releases/tag/M119.0.6045.214) works.

@ladyrick There is a new workaround in #67 That one seems to be most active. I would advise closing this issue as a duplicate.

**Comment by Alex313031** [comment]
@ladyrick @anta40 @gothickitty93 @matgessel @Drugoy @melMass 
Two days ago I got a MASSIVE influx of issues about the MacOS build suddenly crashing.

ITS NOT Thorium's fault. A MacOS update borked Chromium M129 and earlier. Confirmed by making some plain vanilla builds of Chromium as well as downloading official builds from woolyss.com, and they all crashed on MacOS if they were older than M130.

The problem is, I'm trying to find where they fixed it recently, and backport the fix to Thorium M128. The issue is FINDING THE FIX among the hundreds of commits that are made each day, spread across thousands of files, to the repo.

Me and @midzer are trying to fix this, because I don't want to have yet ANOTHER release where MacOS is behind.
